doc/add_function.md

```diff
@@ -10,153 +15,263 @@
-> That means that **_it should be fast_**, that any error in it will put the cell to `#ERROR`.
+  The `UMINUS` function below preserves its argument format, be it a cell reference
+  format (`=UMINUS(A1)`) or the return format of another function (`=UMINUS(CEILING(1.2))`)
 
-```javascript
-const NEW_FORMULA : {
+```ts
+const UMINUS = {
+  description: "A number with the sign reversed.",
+  args: args(`
+    value (number) The number to have its sign reversed. Equivalently, the number to multiply by -1.
+  `),
+  computeFormat: (value) => value?.format,
+  returns: ["NUMBER"],
+  compute: function (value) {
+    return -toNumber(value);
+  },
+};
+```
+
+### Argument definition
+
+Arguments are declared using a string-based syntax `"<arg name> (<arg type>, <other attributes>) <description>"`
+which is interpreted by the `args` function.
+
+```ts
+const { args } = spreadsheet.helpers.args; // get the args function
+
+const MY_FUNC = {
+  args: args(`
+    first_param (string) description of first parameter
+    second_param (boolean, optional, default=false) description of second parameter
+  `),
+  compute: function (firstParam, secondParam) {
+    ...
+  }
   ...,
-  compute : function (parameters)  {
-    if (parameters === 0) {
-      // throwing errors allows you to put the cell in `#ERROR` with  you specified error message
-      // the special string [[FUNCTION_NAME]] will be replaced by the actual function name before showing
-      // it to the user, so you can define utility functions and reuse them to validate arguments.
-      throw new Error ("function [[FUNCTION_NAME]] doesn't want the parameters to be 0")
-    }
+};
+```
 
-    // this.env contains the (OWL) environment of the spreadsheet.
-    const env = this.env;
+The table below shows all attributes.
 
-    // this.getters contains all the getters defined in all the plugins of o_spreadsheet,
-    // and those you added yourself
-    const spreadSheetInformation = this.getters;
+| property                  | type                       |                                                                           |
+| ------------------------- | -------------------------- | ------------------------------------------------------------------------- |
+| `type`                    | [ArgType](#argument-types) | the argument type                                                         |
+| `optional`                | boolean (default=false)    | defines that a parameters is optional                                     |
+| `repeating`               | boolean (default=false)    | when you want your formula to accept multiple parameters of the same type |
+| [`lazy`](#lazy-arguments) | boolean (default=false)    | this parameter will not be evaluated until it is accessed                 |
+| `default`                 | any                        | default value of a parameter if it is not defined                         |
 
```

Comment:
Maybe indicate that there is a direction in which to enter the arguments with their option, otherwise the compilation of the functions returns an error.

- first the arguments without options 
- then the optional arguments or default value
- then the repeatable arguments